### PR TITLE
Convert Autocomplete, ResultList, and SearchInput to JSX

### PR DESCRIPTION
**Pull request by darkwing**
#3506

### CODE REVIEW

File: src/components/shared/SearchInput.js [hunk 1]
```diff
@@ -101,31 +107,31 @@ class SearchInput extends Component {
...
+      onFocus,
+      onBlur,
+      placeholder,
+      value: query,
+      spellCheck: false,
+      ref: "input"
+    };
+
+    return (
+      <div className="search-field {size || ''}">
```
Comment (jasonLaster) [inline src/components/shared/SearchInput.js]: shouldn't it be this: `{`"search-field ${size || ''}"`}`

I'm still new at this game

**Comment by darkwing** [comment]
Good shout; updated!

File: src/components/shared/SearchInput.js (hunk 1 above)
Comment (wldcordeiro) [inline src/components/shared/SearchInput.js]: `<div className={classnames("search-field", { size })}>` would be even cleaner.

**Comment by darkwing** [comment]
This PR keeps choking on an old snapshot; how can I generate a new one for the updated components?

**Comment by zaggy** [comment]
@darkwing `jest --updateSnapshot`

**Comment by wldcordeiro** [comment]
@zaggy that option is also available as `-u` 😄 and you can have yarn run it for you with `yarn test -- -u`

**Comment by darkwing** [comment]
Tests passing with snapshot update, should be good to go

**Pull request merged**